### ISSUE TITLE
Document Pod Density and Default Instance Profile

### ISSUE THREAD

**Issue by prateekgogia** [aws/karpenter#1216]
**Is an existing page relevant?**

**What karpenter features are relevant?**

https://github.com/aws/karpenter/pull/1032
https://github.com/aws/karpenter/pull/914

**How should the docs be improved?**
For the issues linked above, Karpenter is missing docs update

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by mikesir87** [comment]
I'd be happy to take this on, since I authored the two mentioned PRs.

@ellistarn We don't have a good spot in the docs for CLI flags. Add a new page or add it to an existing one? In this case, both are AWS-specific too. Maybe a single CLI flag page with different sections for the various providers?

**Comment by ellistarn** [comment]
@geoffcline @chrisnegus can you chime in on structure?

**Comment by geoffcline** [comment]
Suggestion: Add "reference" section, move provisioner API under new section; add karpenter args page to this section; identify AWS specific content using a "☁️ AWS" header. This is assuming the page would be <50% AWS specific content.

**Comment by alekc** [comment]
Since we are at it, I would suggest adding a separate value in helm chart for this env var, something like 
```yaml
aws:
  # -- The default instance profile to use when launching nodes on AWS
  defaultInstanceProfile: ""
  eniLimitedPodDensity: true
```
which if set to **false** will add envvar to the controller deployment

**Comment by geoffcline** [comment]
@ellistarn can I get a 2-3 sentence summary of https://github.com/aws/karpenter/pull/1032 and this issue?

**Comment by ellistarn** [comment]
Nodes are configured support a maximum number of pods, specific to the cloud provider. Previously for AWS, this number was limited by the number of attachable ENIs on the specific EC2 instance type. With AWS VPC CNI v1.9+, much higher pod densities are supported. By default, `--aws-eni-limited-pod-density=false`, the AWS cloud provider will limit the number of pods to the default Kubernetes upstream value of 110.

**Comment by alekc** [comment]
@ellistarn 

> By default, --aws-eni-limited-pod-density=false, the AWS cloud provider will limit the number of pods to the default Kubernetes upstream value of 110.

I might be misreading what you wrote, but the `--aws-eni-limited-pod-density` is true by default (https://github.com/aws/karpenter/blob/5a5d01572480f54f7c0aad1f0238ec078a958319/pkg/utils/options/options.go#L43), thus limiting pod capacity to 11 pods in case of t3.small

How about something along the lines "To support higher density, set `--aws-eni-limited-pod-density=true` (or env var `AWS_ENI_LIMITED_POD_DENSITY=true`) and change your cni configuration as per aws documentation https://docs.aws.amazon.com/eks/latest/userguide/cni-increase-ip-addresses.html

**Comment by geoffcline** [comment]
closed with #1237
